dbms/src/Columns/ColumnString.cpp

```diff
@@ -596,116 +919,168 @@ void ColumnString::serializeToPosForColumnArrayImpl(
         array_offsets.back(),
         size());
 
-    /// countSerializeByteSizeForColumnArray has already checked that the size of one element is not greater than UINT32_MAX
-    for (size_t i = 0; i < length; ++i)
+    /// countSerializeByteSizeForCmpColumnArray has already checked that the size of one element is not greater than UINT32_MAX
+    if constexpr (has_collator)
```

Comment:
two `for (size_t i = 0; i < length; ++i)` can be merged like `serializeToPosImpl`.